packages/x-components/src/x-modules/facets/components/preselected-filters.vue

```diff
@@ -33,16 +33,42 @@
     public filters!: string[];
 
     /**
-     * Emits the provided preselected filters prioritizing the {@link SnippetConfig} over the
+     * Gets the provided preselected filters prioritizing the {@link SnippetConfig} over the
      * filters prop.
+     *
+     * @returns An array of filter's ids.
      */
-    created(): void {
-      const preselectedFilters = this.snippetConfig?.filters ?? this.filters;
-      if (!isArrayEmpty(preselectedFilters)) {
-        this.$x.emit('PreselectedFiltersProvided', createRawFilters(preselectedFilters));
+    protected get preselectedFilters(): string[] {
+      return this.snippetConfig?.filters ?? this.filters;
+    }
+
+    /**
+     * Emits the {@link FacetsXEvents.PreselectedFiltersProvided} to save
+     * the provided filters in the state.
+     */
+    protected emitPreselectedFilters(): void {
```

Comment:
You can put the watcher here and still call the function independently

```suggestion
    @Watch('preselectedFilters')
    protected emitPreselectedFilters(): void {
```